autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -89,6 +87,47 @@ fn timestamp_filter() -> Result<data::RowFilter, error::BigTableError> {
     Ok(timestamp_filter)
 }
 
+/// Return a RowFilter limiting to a match of the specified `version`'s column
+/// value
+fn version_filter(version: &Uuid) -> data::RowFilter {
+    let mut router_filter_chain = RowFilter_Chain::default();
+    let mut filter_set: RepeatedField<RowFilter> = RepeatedField::default();
+
+    let mut family_filter = data::RowFilter::default();
+    family_filter.set_family_name_regex_filter(format!("^{ROUTER_FAMILY}$"));
+
+    let mut cq_filter = data::RowFilter::default();
+    cq_filter.set_column_qualifier_regex_filter("^version$".as_bytes().to_vec());
+
+    let mut value_filter = data::RowFilter::default();
+    value_filter
+        .set_value_regex_filter(format!("^{}$", version.as_hyphenated()).as_bytes().to_vec());
+
+    filter_set.push(family_filter);
+    filter_set.push(cq_filter);
+    filter_set.push(value_filter);
+
+    router_filter_chain.set_filters(filter_set);
+    let mut router_filter = RowFilter::default();
+    router_filter.set_chain(router_filter_chain);
+    router_filter
+}
+
+/// Return a newly generated `version` column `Cell`
+fn new_version_cell(timestamp: SystemTime) -> cell::Cell {
+    let value = Uuid::new_v4()
+        .as_hyphenated()
```

Comment:
I'm curious why you want this value to be hyphenated and not simple?

Reply:
Happy to switch to simple, I don't really care (I even attempted to store it as a more compact type but I ran into problems with the regex_filter'ing, maybe I wasn't using it right).

Reply:
switched it to binary